File: bindings/node/examples/capture.ts
import { Offprint } from "offprint";

const offprint = new Offprint();

try {
  const result = await offprint.capture("https://example.com", {
    output: "example.html",
  });
  if (result.artifact.kind !== "file") {
    throw new Error("expected a file artifact");
  }
  console.log(result.artifact.path);
} finally {
  await offprint.close();
}

File: bindings/python/examples/capture.py
import asyncio

from offprint import Offprint


async def main() -> None:
    async with Offprint() as offprint:
        result = await offprint.capture(
            "https://example.com",
            output="example.html",
        )
        print(result["artifact"]["path"])


asyncio.run(main())
